src/components/ChatInput/chatinput.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+//images
+import cameraIcon from "../../assets/images/Camera.svg";
+import appStoreIcon from "../../assets/images/App-Store.svg";
+import dictationIcon from "../../assets/images/Dictation.svg";
+import sendIcon from "../../assets/images/sendMessage.svg";
+//채팅 입력받는 component
+
+interface ChatInputProps {
+  onSend: (message: string) => void;
+}
+
+const ChatInput: React.FC<ChatInputProps> = ({ onSend }) => {
+  const [inputValue, setInputValue] = useState<string>("");
+  const [isInputFocused, setIsInputFocused] = useState<boolean>(false);
+
+  //여러줄 입력 위해 input=> textarea로 변경
+  const handleInputEnter = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    if (e.key === "Enter" && !e.shiftKey) {
+      e.preventDefault();
+      if (inputValue.trim() !== "") {
+        // 입력값이 공백인 경우 예외 처리
+        onSend(inputValue);
+      }
+      // 입력값 초기화
+      setIsInputFocused(false);
+      setInputValue("");
+    }
+    if (e.key === "Enter" && e.shiftKey) {
+      e.preventDefault();
+      setInputValue((prev) => prev + "\n");
+    }
+  };
+
+  const handleSendClick = () => {
+    console.log("Send button clicked"); // Add this line for debugging
+    if (inputValue.trim() !== "") {
+      onSend(inputValue);
+    }
+    setInputValue("");
+  };
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setInputValue(e.target.value);
+    setIsInputFocused(e.target.value !== ""); // isInputFocused true로 설정
+  };
+
+  return (
+    <BottomBarContainer>
+      {!isInputFocused && <Camera src={cameraIcon} alt="camera Icon" />}
+      {!isInputFocused && <AppStore src={appStoreIcon} alt="camera Icon" />}
+
+      <InputContainer>
+        <Input
+          autoFocus
+          placeholder="Send Messages..."
+          value={inputValue}
+          onChange={handleInputChange}
+          onKeyPress={handleInputEnter}
+          //한글 입력의 경우: 마지막 단어가 input으로 들어가는 에러 방지하기 위해
+          onFocus={() => setIsInputFocused(true)}
+          onBlur={() => setIsInputFocused(false)}
+          required
+        />
+        {isInputFocused ? (
+          //handleSendClick 작동?? click 작동
+          <Send onClick={handleSendClick} />
```

Comment:
이 부분에서 onClick 보다 onBlur가 우선순위를 가지면서 클릭되기 전에 `<Send>` 가 사라져 클릭으로 메세지가 안 보내지는 이슈가 있는거 같아요. 저 같은 경우에는 onMouseDown으로 해결했습니다! 한번 읽어보시면 좋을거 같아요 ㅎㅎ
```suggestion
        {isInputFocused ? (
          //handleSendClick 작동?? click 작동
          <Send onMouseDown={handleSendClick} />
```
[참고자료](https://kang-ju.tistory.com/5)

Reply:
헉 그렇군요!!! 사실 계속 안돼서 구현을 못했던 부분인데, 참고해서 수정해야겠네요~!!! 알려주셔서 감사합니다~~🙇‍♀️ㅎㅎㅎ